test/metabase/query_processor_test/share_test.clj

```diff
@@ -32,14 +32,22 @@
                                    [:ends-with $name "t"]]]]]})))))
 
     (testing "empty results"
-      ;; due to a bug in the Mongo counts are returned as empty when there are no results (#5419)
-      (is (= (if (= driver/*driver* :mongo)
-               []
-               [[nil]])
-             (mt/rows
-               (mt/run-mbql-query venues
-                 {:aggregation [[:share [:< $price 4]]]
-                  :filter      [:> $price Long/MAX_VALUE]})))))))
+      ;; Vertica doesn't allow dividing null by zero
+      ;; TODO consider wrapping all divisions in nullif checking the first argument
```

Comment:
Is it worth wrapping every division to be compatible with other DBs?

Reply:
I don't think most other databases support dividing by zero. So if we're running into divide by zero errors it's probably a symptom of something else. `[:> $price Long/MAX_VALUE]` should mean that no rows match, so doing a `:share` aggregation over no rows shouldn't even be trying to divide anything

Reply:
Well, doesn't the fact that the other databases in `(disj (mt/normal-drivers-with-feature :basic-aggregations) :mongo)` pass this test imply that most DBs support `null / 0` (and return `null`)?